iothub_client/src/iothubtransport_amqp_common.c

```diff
@@ -467,24 +412,36 @@ static DEVICE_MESSAGE_DISPOSITION_RESULT get_device_disposition_result_from(IOTH
 
 static DEVICE_MESSAGE_DISPOSITION_RESULT on_message_received(IOTHUB_MESSAGE_HANDLE message, DEVICE_MESSAGE_DISPOSITION_INFO* disposition_info, void* context)
 {
-    AMQP_TRANSPORT_DEVICE_INSTANCE* amqp_device_instance = (AMQP_TRANSPORT_DEVICE_INSTANCE*)context;
     DEVICE_MESSAGE_DISPOSITION_RESULT device_disposition_result;
-    MESSAGE_CALLBACK_INFO* message_data;
+    MESSAGE_DISPOSITION_CONTEXT* dispositionContext;
```

Comment:
nit: can this be called 
```suggestion
    MESSAGE_DISPOSITION_CONTEXT* dispositionContextClone;
```
makes it a little easier to follow and understand what is being destroyed and set

Reply:
Sure. Done.